update/ansible/playbook/tasks/roles/postgres/tasks/main.yml

```diff
@@ -217,20 +217,38 @@
         state: absent
       when: is_upgrade.stat.exists
 
-    - name: Reread supervisord configuration
+    - name: Reread supervisord configuration EL7
+      when: ansible_distribution == 'CentOS' and ansible_distribution_major_version == '7'
+      command: supervisorctl reread
+      register: reread_result
+      changed_when: "'No config updates to processes' not in reread_result.stdout"
+
+    - name: Reread supervisord configuration EL9
+      when: (ansible_distribution == 'OracleLinux' or ansible_distribution == 'AlmaLinux') and ansible_distribution_major_version == '9'
       command: /usr/local/bin/supervisorctl reread
       become: true
-      when: is_upgrade.stat.exists
+      register: reread_result
+      changed_when: "'No config updates to processes' not in reread_result.stdout"
+
+    - name: Restart Postgres | EL7
+      command: supervisorctl {{ item }} postgresql
+      changed_when: True
+      become: true
+      loop:
+        - stop
+        - remove
+        - add
+      when: is_upgrade.stat.exists and ansible_distribution == 'CentOS' and ansible_distribution_major_version == '7'
 
-    - name: Restart Postgres
+    - name: Restart Postgres | EL9
       command: /usr/local/bin/supervisorctl {{ item }} postgresql
       changed_when: True
       become: true
       loop:
         - stop
         - remove
         - add
-      when: is_upgrade.stat.exists
+      when: is_upgrade.stat.exists and (ansible_distribution == 'OracleLinux' or ansible_distribution == 'AlmaLinux') and ansible_distribution_major_version == '9'
```

Comment:
```suggestion
      when: 
        - is_upgrade.stat.exists
        - ansible_distribution == 'OracleLinux' or ansible_distribution == 'AlmaLinux'
        - ansible_distribution_major_version == '9'
```

Reply:
nit, but much easier to read

Reply:
I'm not an expert in Ansible, I just copied from other examples we have in ansible scripts that we have :)